cs/testing-with-travis.texy

```diff
@@ -0,0 +1,208 @@
+
+/--
+matrix:
+  exclude:
+    - php: 5.3
+      env: TESTER_PHP_BIN="hhvm"
+
+    - php: 5.4
+      env: TESTER_PHP_BIN="hhvm"
+
+    - php: 5.5
+      env: TESTER_PHP_BIN="hhvm"
+
+    - php: 5.6
+      env: TESTER_PHP_BIN="hhvm"
+
+    - php: hhvm
+      env: TESTER_PHP_BIN="php"
+\--
+
+Aby testování bylo označeno Travisem jako úspěšné, musí projít všechny dílčí testy. Pokud nechceme, aby výsledek některé z nich ovlivňovaly, můžeme to deklarovat v bloku `allow_failures`. V našem případě nám nebude vadit, pokud neprojdou testy na HHVM.
+
+/--
+matrix:
+  allow_failures:
+    - php: hhvm
+\--
+
+Pokud nyní testy pod HHVM selžou, ale pod jinými verzemi projdou, bude build označen jako procházející.
+
+Spouštění testů
+===============
+
+Testy se spouštějí v sekci `script`, kde stačí zavolat tester. Předpokládejeme, že testy jsou umístěny ve složce `tests/` a používáte vlastní `php.ini` umístěné ve složce s testy.
+
+/--
+script:
+  - ./vendor/bin/tester -p $TESTER_PHP_BIN -c ./tests/php.ini ./tests
+\--
+
+Když se něco pokazí
+===================
+
+Travis nám umožňuje nastavit i blok `after_failure`, který se spustí v případě, že test selže. Pomocí tohoto bloku si můžeme například nechat vypsat výstupy funkcí, které se lišily od očekávaných výstupů uvedených v testech.
+
+/--
+after_failure:
+    # Vytiskne obsah souborů *.actual
+    - for i in $(find tests -name \*.actual); do echo "--- $i"; cat $i; echo; echo; done
+\--
+
+Nastavení dalších služeb
+========================
+
+Travis nám umožňuje zapnout další služby. Například Redis server na který ukládáte cache.
+
+/--
+services:
+  - redis-server
+\-- 
+
+Inicializace databáze:
+======================
+
+Databáze MySQL je již předinstalována. Běží na adrese `127.0.0.1` a přihlásit se můžete na účet `travis` nebo `root` bez hesla. Pokud byste chtěli spouštět integrační testy, můžete si do ni naimportovat testovací databázi, kterou máte například v souboru `tests/testbase.sql`:
+
+/--
+before_script:
+  - mysql -u root -e 'CREATE DATABASE testbase;'
+  - mysql -u root testbase < tests/testbase.sql
+\--
+
+Výsledný soubor
+===============
+
+Celý `.travis.yml` pak vypadá takto:
+
+/--
+language: php
+
+php:
+  - 5.3
+  - 5.4
+  - 5.5
+  - 5.6
+  - hhvm
+
+env:
+  - TESTER_PHP_BIN="php"
+  - TESTER_PHP_BIN="hhvm"
+
+matrix:
+  allow_failures:
+    - php: hhvm
+
+  exclude:
+    - php: 5.3
+      env: TESTER_PHP_BIN="hhvm"
+
+    - php: 5.4
+      env: TESTER_PHP_BIN="hhvm"
+
+    - php: 5.5
+      env: TESTER_PHP_BIN="hhvm"
+
+    - php: 5.6
+      env: TESTER_PHP_BIN="hhvm"
+
+    - php: hhvm
+      env: TESTER_PHP_BIN="php"
+
+services:
+  - redis-server
+
+before_install:
+  - composer self-update
+
+install:
+  - composer install --no-interaction --prefer-source --dev
+
+before_script:
+  - mysql -u root -e 'CREATE DATABASE testbase;'
+  - mysql -u root testbase < tests/testbase.sql
+
+script: ./vendor/bin/tester -c ./tests/php.ini -p $TESTER_PHP_BIN ./tests/
```

Comment:
Nekonzistentní formátování, hodil bych to taky jako odrážku, tedy

~~~
script:
    - ./vendor/bin/tester -c ./tests/php.ini -p $TESTER_PHP_BIN ./tests/
~~~